### ISSUE TITLE
2001 - Null response from service Centinel

### ISSUE THREAD

**Issue by naxel** [braintree/braintree_ios#771]
**Integration Details:**
 - Braintree SDK Version: 5.5.0
 - Environment: Production
 - iOS Version: 15.1
 - Xcode Version: 13.2.1
 - Device: iPhone XR
 - Integration type & version: CocoaPods 1.11.2
 - Development Processor Type: Intel

**Describe the bug**
Some of our users can’t make payments in our App. On the Security Checkout screen, after pressing the button “Fortsatt” (“Continue”), it is immediately finished the 3DS verification.
After debugging, I have found what Braintree SDK send request:
```
POST https://centinelapi.cardinalcommerce.com/V1/Order/JWT/StepUp
```
and the response contains:
```
“Payload”: {
    “Successful”: false,
    “ErrorNumber”: 2001,
    “ErrorDescription”: “Null response from service Centinel”
  }
```
Why does it happen, and what does it mean?
Is it possible to handle this error?
Thank you!

**Screenshots**
![2021-12-30 15 10 48](https://user-images.githubusercontent.com/1148505/147756349-b7f9dafa-b2ac-4344-bd83-04e481064b3a.jpg)

**Comment by sshropshire** [comment]
Hi @naxel thanks for using the Braintree SDK for iOS. This looks like an issue with Cardinal our 3rd-party MPI plugin. Do you have a set of steps to reproduce this issue?

**Comment by naxel** [comment]
Hi @sshropshire, unfortunately, I can't provide stable steps to reproducing. 
It is only sometimes reproducing. 
For the same card, user info, the same phone, IP, and location it sometimes reproduces and sometimes not. I noticed when it reproduces it is showing the Secure Checkout screen, but when is not reproducing it is charging money without showing that screen.
It seems it is more often happened for some cards than for others.

**Comment by sarahkoop** [comment]
We've reported this issue to Cardinal, and they've confirmed this is known issue on iOS, and they are actively working on a fix. We will post an update here when the fix has been released.

**Comment by naxel** [comment]
thank you

**Comment by mxkh** [comment]
Hello @sarahkoop, we appreciate your help. I hope that your partners will fix this issue as soon as possible, as it often occurs and causes some inconvenience with payments for our users. Our support team receives at least one ticket per day about this issue.

Many thanks for considering our request.

cc @naxel

**Comment by Tobbb** [comment]
hello @sarahkoop , is there any update on this issue?

**Comment by sarahkoop** [comment]
Hi all - we have reached out to Cardinal for another update, and will post here when we have more information. We appreciate your patience.

**Comment by naxel** [comment]
Hi @sarahkoop, any news?

**Comment by sarahkoop** [comment]
We are expecting this to be included in the next Cardinal release! We will update the SDK with the new version as soon as possible, once it's released.

**Comment by jaxdesmarais** [comment]
Hello @Tobbb and @naxel -

We've been working with the Cardinal team closely on this issue. I wanted to provide a quick update as there are 2 issues at play:
1. The debug response "Null response from service Centinel" in the console. This issue is _not_ resolved but it is not indicative of the issue at play. Cardinal is sending this response a majority of the time even on transactions that are fully successful. They are working to determine why they are sending this incorrect messaging, but this messaging can be disregarded.
2. The issue where some customers are seeing the incorrect flow for 3D Secure. Cardinal believes to have fixed this issue in the newest version of their SDK. We have released our SDK with this new version of their SDK in Braintree SDK version 5.10.0.

For the second issue, please pull in the latest version of the SDK with this fix included and let us know the results. This should resolve the second issue some users were seeing.

**Comment by naxel** [comment]
Hello @sarahkoop and @jaxdesmarais
do you have any news about this issue?
Thanks

**Comment by jaxdesmarais** [comment]
Hello @naxel -

Thanks for reaching back out. Are you able to have one of your customers capture a video of the behavior they are seeing? That will allow us to pass on these details to Cardinal to investigate further as they had indicated this would be fixed in the Cardinal version released in version 5.10.0 of the Braintree SDK.

**Comment by hollabaq86** [comment]
Hi folks,

**If you are on the latest version** of the Braintree SDK and **continue to encounter either of the issues** mentioned by @jaxdesmarais:
> 1. The debug response "Null response from service Centinel" in the console. This issue is not resolved but it is not indicative of the issue at play. Cardinal is sending this response a majority of the time even on transactions that are fully successful. They are working to determine why they are sending this incorrect messaging, but this messaging can be disregarded.
> 2. The issue where some customers are seeing the incorrect flow for 3D Secure. Cardinal believes to have fixed this issue in the newest version of their SDK. We have released our SDK with this new version of their SDK in Braintree SDK version 5.10.0.

**Please [contact Support](https://developer.paypal.com/braintree/help)** referencing this GH issue, and try to provide as much of the following information as you can to help us help Cardinal investigate their logs.

- **timestamps** (preferably in UTC, but honestly just let us know the timezone) of when the issue occured
- **amount** that was attempted for 3D Secure verification
- (if possible to get) the **last 4 digits** of the user's BIN that attempted 3D Secure verification.

Since we have not heard back from folks in this issue for a few weeks, I'm going to go ahead and close this issue for inactivity.
